packages/cli/src/io.ts

```diff
@@ -208,36 +208,36 @@ export async function outputCodegen(
     )}} from "@polkadot-api/substrate-bindings"`,
   )
 
+  const tscFileName = path.join(outputFolder, key)
+  const tscTypesFileName = path.join(outputFolder, `${key}-types`)
```

Comment:
Have you tried this https://github.com/microsoft/TypeScript/wiki/Using-the-Compiler-API#getting-the-dts-from-a-javascript-file?

I'd rather use `typescript` directly instead of adding `tsc-prog`

WDYT?

Reply:
Resolved together offline